moped-editor/src/utils/mapSaveActionReducer.js

```diff
@@ -0,0 +1,81 @@
+/* Save Action State Reducer */
+export const mapSaveActionInitialState = () => ({
+  currentStep: 0,
+  initiateFeatureSave: false, // Tells the map to save all features
+  featuresSaved: false, // Tells the component editor to save the component
+  componentSaved: false, // Tells the world the component is saved
+  exit: false, // Tells the world it's time to exit
+  errors: null,
+  message: null,
+});
```

Comment:
This is the initial state of our reducer, it begins at step zero (out of four), which controls the "save" component sequence. First it tells the map to save all its features with "initiateFeatureSave", then the map sends a message back whenever it is done via "featuresSaved", then the map's parent will attempt to save the component and if successful it will send the signal "componentSaved", once the signal is received it also sends the "exit" signal to go back to the components list.